### PR TITLE
Lots of new annotation and Mutect filters. Closes #964, #965.

### LINKED ISSUE #964: M2 read position filter
Implement a filter based on the median alt position annotation.

### PR DESCRIPTION
**Pull request by davidbenjamin**
@LeeTL1220 This will put the new filters used in our most recent evaluations into master.

@takutosato I'm giving this to Lee because I'm about to assign the new somatic likelihoods PR to you for review, but it's worth acquainting yourself with the new filters we have.  

Note that some are not yet at all useful, but I'm anticipating that *maybe* they will be good covariates as part of an ML model.

### CODE REVIEW

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/ClippedBases.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+
+import java.util.List;
+import java.util.OptionalDouble;
+import java.util.OptionalInt;
+
+/**
+ * Created by David Benjamin on 3/20/17.
+ */
+public class ClippedBases extends PerAlleleAnnotation {
```
Comment (davidbenjamin) [inline src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/ClippedBases.java]: done

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/ClippedBases.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    private static byte BASE_QUAL_THRESHOLD = 25;
+
+    public static final String KEY = "MCL";
```
Comment (davidbenjamin) [inline src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/ClippedBases.java]: Yes, but that class is in gatk public.  Later I will move all of this to `GATKVCFConstants`.

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/PerAlleleAnnotation.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+import org.broadinstitute.hellbender.utils.read.ReadUtils;
+import org.broadinstitute.hellbender.utils.variant.GATKVCFConstants;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * Created by David Benjamin on 4/13/17.
+ */
+public abstract class PerAlleleAnnotation extends GenotypeAnnotation {
```
Comment (davidbenjamin) [inline src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/PerAlleleAnnotation.java]: done

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/PerAlleleAnnotation.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+    public void annotate(final ReferenceContext ref,
```
Comment (davidbenjamin) [inline src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/PerAlleleAnnotation.java]: done

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/Mutect2FilteringEngine.java [hunk 5]
```diff
@@ -81,6 +87,53 @@ private static void applyPanelOfNormalsFilter(final M2FiltersArgumentCollection
         }
     }
 
+    private void applyMedianBaseQualityDifferenceFilter(final M2FiltersArgumentCollection MTFAC, final VariantContext vc, final Collection<String> filters) {
```
Comment (davidbenjamin) [inline src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/Mutect2FilteringEngine.java]: addressed already in this PR; deleted the TODO

**Comment by davidbenjamin** [comment]
@LeeTL1220 Responded to all comments so far.

**Comment by LeeTL1220** [comment]
@davidbenjamin Merge away!

**Pull request merged**